### PR TITLE
Added tests for the React frontend including the bar chart component.

### PR DESCRIPTION
**Pull request by ezhaohang**

### CODE REVIEW

File: risr-app/risr_proj/dashboard-fe/src/App.test.js [hunk 1]
```diff
@@ -15,11 +15,15 @@
...
 import App from './App';
+import Header from './components/Header';
+import Home from './components/Home';
 
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+describe('App component', () => {
+  test('renders without crashing', () => {
```
Comment (eyurko) [inline risr-app/risr_proj/dashboard-fe/src/App.test.js:24]: i suppose that's fair

File: risr-app/risr_proj/dashboard-fe/src/components/BarChart.js [hunk 2]
```diff
@@ -62,9 +62,14 @@ class BarChart extends Component {
...
   static getDerivedStateFromProps(nextProps, prevState) {
     // Check if data has been loaded.
-    if (!nextProps.data) return null;
+    // console.log(nextProps.data === {[]});
+    if (!nextProps.data || !nextProps.data.length) return null;
 
+    // Check if data has the expected keys.
     const {data} = nextProps;
+
+    if (!('pr_range' in data[0]) || !('repo_count' in data[0])) return null;
```
Comment (eyurko) [inline risr-app/risr_proj/dashboard-fe/src/components/BarChart.js:71]: add comment explaining this -- why are pr_range and repo_count required keys?

File: risr-app/risr_proj/dashboard-fe/src/components/BarChart.test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+describe('Bar chart component', () => {
+  test('renders a svg element', () => {
+    const shallowChart = shallow(<BarChart />);
+    expect(shallowChart.find('svg')).toHaveLength(1);
+    expect(shallowChart.find('g')).toHaveLength(3);
+  });
+
+  test('renders expected number of rectangles', () => {
+    const data = [
+      {'pr_range': '0-1', 'repo_count': 136},
```
Comment (eyurko) [inline risr-app/risr_proj/dashboard-fe/src/components/BarChart.test.js:30]: can you make constants for these numbers / comment to explain why they should be these numbers?

Comment (ezhaohang) [reply risr-app/risr_proj/dashboard-fe/src/components/BarChart.test.js:30]: These numbers don't really matter for the test. Would it be alright if I just changed everything to 1?

File: risr-app/risr_proj/dashboard-fe/src/components/BarChart.js (hunk 2 above)
Comment (ezhaohang) [reply risr-app/risr_proj/dashboard-fe/src/components/BarChart.js:71]: At the moment, the bar chart component only works with one data set. Would it be alright if I just add a comment for now and then maybe change the component to work with more data sets later?

Comment (eyurko) [reply risr-app/risr_proj/dashboard-fe/src/components/BarChart.js:71]: sure!

File: risr-app/risr_proj/dashboard-fe/src/components/BarChart.test.js (hunk 3 above)
Comment (eyurko) [reply risr-app/risr_proj/dashboard-fe/src/components/BarChart.test.js:30]: that may make it more confusing

Comment (ezhaohang) [reply risr-app/risr_proj/dashboard-fe/src/components/BarChart.test.js]: Hm, I modified the test a bit to also check for height and now all the constants are 10. What do you think? Confusing?

**Pull request merged**